[SPMD] Add a dump_graphs_to_files utils to facilitate graph transformation debug

fegin opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #98580
* #98579
* #98578
* #98285
* __->__ #98284
* #98182

Throughout the compilation, there are multiple graphs that will be generated.  This PR add an utils to dump the result graphs to a folder.

Differential Revision: [D44661599](https://our.internmc.facebook.com/intern/diff/D44661599/)

fegin committed:
[SPMD] Add a dump_graphs_to_files utils to facilitate graph transformation debug

Throughout the compiling, there are multiple graphs that will be generated.  This PR add an utils to dump the result graphs to a folder.

Differential Revision: [D44661599](https://our.internmc.facebook.com/intern/diff/D44661599/)

[ghstack-poisoned]

mrshenli commented on torch/distributed/_spmd/graph_utils.py:
```diff
@@ -23,60 +29,6 @@ class CommType(str, Enum):
-)
-
-
-def get_comm_block_nodes(
```
nit: split this out as a separate PR?

mrshenli commented on torch/distributed/_spmd/graph_utils.py:
```diff
@@ -166,3 +118,16 @@ def rebuild_graph(gm: fx.GraphModule, remove_dead_code: bool = True) -> None:
     gm.recompile()
+
+
+def dump_graphs_to_files(graphs: Dict[str, fx.GraphModule], folder: str = "") -> str:
```
nit: do we need tests for this?

fegin replied:
Done

fegin committed:
Update on "[SPMD] Add a dump_graphs_to_files utils to facilitate graph transformation debug"


Throughout the compilation, there are multiple graphs that will be generated.  This PR add an utils to dump the result graphs to a folder.

Differential Revision: [D44661599](https://our.internmc.facebook.com/intern/diff/D44661599/)

[ghstack-poisoned]

fegin committed:
Update on "[SPMD] Add a dump_graphs_to_files utils to facilitate graph transformation debug"


Throughout the compilation, there are multiple graphs that will be generated.  This PR add an utils to dump the result graphs to a folder.

Differential Revision: [D44661599](https://our.internmc.facebook.com/intern/diff/D44661599/)

[ghstack-poisoned]

fegin committed:
Update on "[SPMD] Add a dump_graphs_to_files utils to facilitate graph transformation debug"


Throughout the compilation, there are multiple graphs that will be generated.  This PR add an utils to dump the result graphs to a folder.

Differential Revision: [D44661599](https://our.internmc.facebook.com/intern/diff/D44661599/)

[ghstack-poisoned]

fegin committed:
Update on "[SPMD] Add a dump_graphs_to_files utils to facilitate graph transformation debug"


Throughout the compilation, there are multiple graphs that will be generated.  This PR add an utils to dump the result graphs to a folder.

Differential Revision: [D44661599](https://our.internmc.facebook.com/intern/diff/D44661599/)

[ghstack-poisoned]

fegin committed:
Update on "[SPMD] Add a dump_graphs_to_files utils to facilitate graph transformation debug"


Throughout the compilation, there are multiple graphs that will be generated.  This PR add an utils to dump the result graphs to a folder.

Differential Revision: [D44661599](https://our.internmc.facebook.com/intern/diff/D44661599/)

[ghstack-poisoned]

fegin commented:
Error message of `windows-binary-libtorch-debug / libtorch-cpu-shared-with-deps-debug-test (push)`
```
Copy-Item : Cannot find path 'C:\Users\circleci\AppData\Local\Temp\vslogs.zip' because it does not exist.
At C:\actions-runner\_work\pytorch\pytorch\builder\windows\internal\vs2022_install.ps1:54 char:5
+     Copy-Item -Path "C:\Users\circleci\AppData\Local\Temp\vslogs.zip" ...
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (C:\Users\circle...Temp\vslogs.zip:String) [Copy-Item], ItemNotFoundExce 
   ption
    + FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.CopyItemCommand
```

The error does not seem to be related.

fegin commented:
@pytorchbot merge -f "The failing test is not related, seem the conversation."
